lib/findbugs_translate_checkstyle_format/translate.rb

```diff
@@ -9,46 +10,48 @@ def parse(xml)
     end
 
     def trans(xml)
-      require 'rexml/document'
       doc = REXML::Document.new
       doc << REXML::XMLDecl.new('1.0', 'UTF-8')
 
-      checkstyle = doc.add_element("checkstyle")
-      if xml['BugCollection']['BugInstance']
-        bugInstances = xml['BugCollection']['BugInstance'].is_a?(Array) ? xml['BugCollection']['BugInstance'] : [xml['BugCollection']['BugInstance']]
-        bugInstances.each do |bugInstance|
-          file = checkstyle.add_element("file", {
-            'name' => fqcn_to_path(bugInstance['SourceLine']['@classname'], xml)
-            })
-          file.add_element("error", {
-            'line' => bugInstance['SourceLine']['@start'],
-            'severity' => '',
-            'message' => "[#{bugInstance['@category']}] #{bugInstance['LongMessage']}"
-            })
-        end
-      else
-        # create dummy
-        dummy_src_dir = xml['BugCollection']['Project']['SrcDir']
-        if dummy_src_dir.is_a?(Array)
-          dummy_src_dir = dummy_src_dir.first
-        end
-        file = checkstyle.add_element("file", {
-          'name' => dummy_src_dir
-          })
+      checkstyle = doc.add_element('checkstyle')
+      bug_instances = xml['BugCollection']['BugInstance']
+      if bug_instances.blank?
+        FindbugsTranslateCheckstyleFormat::Translate.set_dummy(xml, checkstyle)
+        return doc
+      end
+
+      bug_instances = [bug_instances] if bug_instances.is_a?(Hash)
+      bug_instances.each do |bug_instance|
+        source_line = bug_instance['SourceLine']
+        file = checkstyle.add_element('file',
+                                      'name' => fqcn_to_path(source_line['@classname'], xml)
+                                     )
+        file.add_element('error',
+                         'line' => source_line['@start'],
+                         'severity' => '',
+                         'message' => "[#{bug_instance['@category']}] #{bug_instance['LongMessage']}"
```

Comment:
__[reek]__
* [FeatureEnvy] refers to bug_instance more than self
* [FeatureEnvy] refers to bug_instance more than self